### PR TITLE
Define classes, person, student, teacher

### PR DESCRIPTION
**Pull request by huseyinbiyik**
You can find the features added for this Pull Request below.

- Class Person created with the following:
1. Instance vars: @id, @name, and @age.
2. Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
3. Getters for @id, @name, and @age.
4. Setters for @name and @age.
5. Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
6. Public method can_use_services? that returns true if person is of age or if they have permission from parents.

- Class Student creted with the following:
1. Inherits from Person.
2. Constructor extends parent's constructor by adding @classroom and a parameter for it.
3. Method play_hooky that returns "¯\(ツ)/¯".

- Class Teacher created with the following:
1. Inherits from Person.
2. Constructor extends parent's constructor by adding @specialization and a parameter for it.
3. Override can_use_services? so it always returns true.

- Each class saved in a separate file.

Thank you for the review in advance!🤗

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  def of_age?
+    @age >= 18
+  end
+  private :of_age?
+
+  def can_use_services?
+    is_of_age? || parent_permission
+  end
```
Comment (thecodechaser) [inline person.rb]: - [ ] On line 18 you are calling the method `is_of_age?` to verify the person's age but the method's correct name is `of_age?` defined on line 12. Please fix this issue by correcting the method name.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @huseyinbiyik👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (huseyinbiyik) [reply person.rb]: Oh, thank you so much for noticing me.

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @huseyinbiyik👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by huseyinbiyik** [comment]
Thank you for the review! I will merge it now. 🎉

**Pull request merged**